targets/bmv2/CMakeLists.txt

```diff
@@ -4,6 +4,17 @@ if(ENABLE_TESTING)
   include(test/P4Tests.cmake)
 endif()
 
+find_program(RTSMITH_DRIVER p4rtsmith)
+if (NOT RTSMITH_DRIVER)
+    message(FATAL_ERROR "RtSmith tool not found! Please ensure it is in your PATH.")
+endif()
+
+# Find p4c (for JSON generation)
+find_program(P4C_EXECUTABLE p4c)
```

Comment:
This is the wrong way to do this, this is why the current tests fail. You directly access the P4C binary in the script, you shouldn't need to use find_program.